core/providers/openai/chat_test.go

```
@@ -80,6 +80,70 @@ func TestToOpenAIChatRequest_ToolNormalization(t *testing.T) {
+	// Setup guard: the case is only meaningful if normalization changes the bytes.
+	require.NotEqual(t, string(staleJSON), string(expJSON),
+		"test setup: normalized form must differ from the cached form")
+
+	bifrostReq := &schemas.BifrostChatRequest{
+		Provider: schemas.OpenAI,
+		Model:    "gpt-4o",
+		Input:    []schemas.ChatMessage{{Role: schemas.ChatMessageRoleUser}},
+		Params:   &schemas.ChatParameters{Tools: []schemas.ChatTool{tool}},
+	}
+
+	ctx, cancel := schemas.NewBifrostContextWithCancel(nil)
```

golangci-lint (2.12.2)

[error] SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

(staticcheck)